kedro-datasets/kedro_datasets_experimental/pandas/delta_sharing_dataset.py

```diff
@@ -0,0 +1,164 @@
+from __future__ import annotations
+
+from typing import Any
+
+import delta_sharing
+import pandas as pd
+from kedro.io.core import AbstractDataset, DatasetError
+
+from kedro_datasets._typing import TablePreview
+
+
+class DeltaSharingDataset(AbstractDataset):
+    """``DeltaSharingDataset`` loads data from a Delta Sharing shared table using the Delta Sharing open protocol.
+    This dataset handles loading data into a Pandas DataFrame. Saving to Delta Sharing is not supported.
+
+    Delta Sharing is an open protocol for secure real-time exchange of large datasets, which enables
+    organizations to share data in real time regardless of which computing platforms they use. It is a
+    simple REST protocol that securely shares access to part of a cloud dataset and leverages modern cloud
+    storage systems, such as S3, ADLS, or GCS, to reliably transfer data.
+
+    Example usage for the YAML API:
+
+    .. code-block:: yaml
+
+        my_delta_sharing_dataset:
+          type: pandas.DeltaSharingDataset
+          share: <share-name>
+          schema: <schema-name>
+          table: <table-name>
+          credentials:
+            profile_file: <profile-file-path>
+          load_args:
+            version: <version>
+            limit: <limit>
+            use_delta_format: <use_delta_format>
+
+    Example usage for the Python API:
+
+    .. code-block:: pycon
+
+        >>> from kedro_datasets import DeltaSharingDataset
+        >>> import pandas as pd
+        >>>
+        >>> credentials = {
+        ...     "profile_file": "conf/local/config.share"
+        ... }
+        >>> load_args = {
+        ...     "version": 1,
+        ...     "limit": 10,
+        ...     "use_delta_format": True
+        ... }
+        >>> dataset = DeltaSharingDataset(
+        ...     share="example_share",
+        ...     schema="example_schema",
+        ...     table="example_table",
+        ...     credentials=credentials,
+        ...     load_args=load_args
+        ... )
+        >>> data = dataset.load()
+        >>> print(data)
+    """
```

Comment:
Is it possible to create an example that can run locally? or is it expected to connect to somewhere?

Reply:
Thank you for your feedback, really appreciate it.

For accessing the Delta Sharing Server, Delta Sharing provides a profile file that allows you to connect to a public example server. You can download the necessary credentials from [this link](https://databricks-datasets-oregon.s3-us-west-2.amazonaws.com/delta-sharing/share/open-datasets.share). More information about the Delta Sharing project can be found [here](https://github.com/delta-io/delta-sharing), their Github repository.

Regarding the runnable example, I’m not sure if you’d prefer it in the code or in the comments, but I’ll provide it here for convenience:

```python
from kedro_datasets_experimental.pandas import DeltaSharingDataset

# Define the credentials for accessing the Delta Sharing profile
credentials = {
    "profile_file": "open-datasets.share"  # Path to the profile file downloaded from link above
}

# Load arguments specifying any additional options for loading data
load_args = {
    "limit": 10,  # Limit the number of rows loaded
}

# Create an instance of the DeltaSharingDataset with the specified parameters
dataset = DeltaSharingDataset(
    share="delta_sharing",  # Name of the share in Delta Sharing
    schema="default",       # Schema name in Delta Sharing
    table="nyctaxi_2019",  # Table name to load data from
    credentials=credentials,  # Pass the credentials for access
    load_args=load_args      # Pass the loading options
)

# Load the data into a Pandas DataFrame
data = dataset.load()
print(data)  # Display the loaded data
```

The public example share includes the following datasets if you'd like to test different examples:
- COVID_19_NYT
- boston-housing
- flight-asa_2008
- lending_club
- nyctaxi_2019
- nyctaxi_2019_part
- owid-covid-data

Please let me know if you need any further information or if you'd like me to incorporate this example directly into the code!

Reply:
I'd prefer this as docstring. Docstring are rendered as doc automatically, for example: https://docs.kedro.org/projects/kedro-datasets/en/kedro-datasets-5.1.0/api/kedro_datasets_experimental.langchain.ChatAnthropicDataset.html

I will try to run this code and convert it to docstring.